src/CMakeLists.txt

```diff
@@ -33,20 +32,15 @@ endif()
 add_samp_plugin(${PROJECT_NAME} ${PLUGIN_SOURCES})
 
 if(WIN32)
-	add_definitions(-DBOOST_ALL_NO_LIB -DNOMINMAX)
+	add_definitions(-DNOMINMAX)
 else()
 	set_property(TARGET ${PROJECT_NAME} APPEND_STRING PROPERTY COMPILE_FLAGS " -fno-strict-aliasing")
 	set_property(TARGET ${PROJECT_NAME} APPEND_STRING PROPERTY LINK_FLAGS " -lrt")
 endif()
 
-add_definitions(-DBOOST_CHRONO_HEADER_ONLY -DSAMPGDK_AMALGAMATION -DSAMPGDK_CPP_WRAPPERS)
+add_definitions(-DSAMPGDK_AMALGAMATION -DSAMPGDK_CPP_WRAPPERS -D_SILENCE_CXX17_NEGATORS_DEPRECATION_WARNING -D_SILENCE_CXX17_ADAPTOR_TYPEDEFS_DEPRECATION_WARNING)
 
 set_target_properties(${PROJECT_NAME} PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin")
-target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
-
-include(cotire)
```

Comment:
Good idea to remove this since it's deprecated. CMake now has native support for [precompiled headers](https://cmake.org/cmake/help/latest/command/target_precompile_headers.html) and [unity builds](https://cmake.org/cmake/help/v3.16/prop_tgt/UNITY_BUILD.html), which would help speed up compilation. Not necessary, but something to consider.